test/Spec/BotPlutusInterface/Config.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Spec.BotPlutusInterface.Config (tests) where
+
+import BotPlutusInterface.Config (loadPABConfig, savePABConfig)
+import BotPlutusInterface.Types (CLILocation (..), LogLevel (..), PABConfig (..))
+import Cardano.Api (
+  AnyPlutusScriptVersion (..),
+  CostModel (..),
+  EpochNo (EpochNo),
+  ExecutionUnitPrices (..),
+  ExecutionUnits (..),
+  Lovelace (..),
+  NetworkId (Mainnet),
+  PlutusScriptVersion (PlutusScriptV2),
+  makePraosNonce,
+ )
+import Cardano.Api.Shelley (ProtocolParameters (..))
+import Data.Default (def)
+import Data.Map qualified as Map
+import Data.Ratio ((%))
+import Data.Text qualified as Text
+import Ledger (POSIXTime (POSIXTime), StakePubKeyHash (StakePubKeyHash))
+import Ledger.TimeSlot (SlotConfig (..))
+import PlutusConfig.Misc (serializeDeserialize)
+import Servant.Client.Core (BaseUrl (..), Scheme (Https))
+import System.FilePath ((</>))
+import System.IO.Temp (withSystemTempDirectory)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase, (@?=))
+import Prelude
+
+{- | Tests for PABConfig serialize/deserialize
+
+ @since 0.1
+-}
+tests :: TestTree
+tests =
+  testGroup
+    "BotPlutusInterface.Config"
+    [ testCase
+        "PABConfig default serialize/deserialize"
+        (serializeDeserialize def @?= Right (def :: PABConfig))
+    , testCase
+        "PABConfig example serialize/deserialize (without pcProtocolParams)"
+        ( serializeDeserialize pabConfigExample
+            @?= Right pabConfigExample {pcProtocolParams = def}
+        )
+    , testCase
+        "PABConfig example serialize/deserialize (without pcProtocolParams)"
+        $ withSystemTempDirectory "PABConfig-test" $ \path -> do
+          let confFile = path </> "conf.value"
+              pcProtocolParamsFile = Text.pack $ path </> "protocol.json"
+              conf = pabConfigExample {pcProtocolParamsFile}
+          savePABConfig confFile conf
+          Right conf' <- loadPABConfig confFile
+          conf' @?= conf
+    ]
+
+pabConfigExample :: PABConfig
+pabConfigExample =
+  PABConfig
+    { pcCliLocation = Remote "localhost"
+    , pcChainIndexUrl = BaseUrl Https "127.0.0.1" 8080 ""
+    , pcNetwork = Mainnet
+    , pcProtocolParams =
```

Comment:
Is there any way we can prevent the protocolParams to be specified like this? Ideally we want to have the field in config be populated by the values in the file, but I can't really see a good reason why we would want to tweak them like this

Reply:
On the haskell level -- I think no. On configuration level -- already.

Here I do it only for testing purposes.